### PR TITLE
Prettier integration

### PR DESCRIPTION
**Pull request by vvscode**
*Issue #, if available:* https://github.com/amzn/style-dictionary/issues/155

*Description of changes:*
- installed `prettier` and eslint plugins to solve possible conflicts
- added `precommit` hook to format `*.js` files on changes
- command which checks diff in prettier formatting is part of `test` script
- code reformatted with prettier

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vvscode** [comment]
@davixyz @chazzmoney

**Comment by chazzmoney** [comment]
I'm new to prettier, so I'm not sure about the options that we can use to manage the code format. I'll spend some time looking at the documentation and see if there are any config changes worth suggesting.

All the other changes look good to me!

File: lib/common/formats.js [hunk 1]
```diff
@@ -11,46 +11,59 @@
...
   return _.map(properties, function(prop) {
-      var to_ret_prop = prefix + prop.name + ': ' + (prop.attributes.category==='asset' ? '"'+prop.value+'"' : prop.value) + ';';
+    var to_ret_prop =
+      prefix +
+      prop.name +
+      ": " +
+      (prop.attributes.category === "asset"
+        ? '"' + prop.value + '"'
+        : prop.value) +
+      ";";
```
[the remark this answers is not in the document]
Comment (chazzmoney) [reply lib/common/formats.js]: Looks like no...

https://github.com/prettier/prettier/issues/3913

**Comment by vvscode** [comment]
@chazzmoney  you could check options at https://prettier.io/playground/

File: lib/common/formats.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (davixyz) [reply lib/common/formats.js]: we could use template literals in all of this so we don't concatenate a bunch of strings, I think node V6 onwards support them out of the box.
https://node.green/

**Comment by chazzmoney** [comment]
Thanks for the link.

The only thing we should for sure do is find a good --print-width

Might want to decide on --no-bracket-spacing

@dbanksdesign @didoo @davixyz Do you have preferences / thoughts on formatting?

**Comment by chazzmoney** [comment]
Also, I should note we could always change it later. As far as just adding the functionality I'm feeling pretty good.

File: .eslintrc.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
     "Buffer": true,
     "escape": true
   },
-  "extends": "eslint:recommended",
+  "extends": ["eslint:recommended", "prettier"],
```
Comment (davixyz) [inline .eslintrc.json]: I know this shouldn't be part of this PR since this is the prettier integration but at some point should we use
`https://www.npmjs.com/package/eslint-config-airbnb-base` for the JS linting? this would allow us to start writing some ES6 supported by the newer versions of node without it complaining about things like consts, etc.

File: lib/common/formats.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (vvscode) [reply lib/common/formats.js]: travis runs tests both for node-8 and node-6

**Comment by davixyz** [comment]
@chazzmoney My preferences on prettier configs are the following:
```
{
  "trailingComma": "es5",
  "singleQuote": true
}
```

I kinda like that there's a trailing comma where we can for ES5 (Max compat) and single quotes; the default of prettier has singleQuotes turned as off and the trailing comma as none.

File: lib/common/formats.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply lib/common/formats.js]: Maybe we can live with it for now, part of the 3.0 should probably be updating our JS to use new es6 stuff which would fix it.

**Comment by vvscode** [comment]
> @chazzmoney My preferences on prettier configs are the following:
> 
> ```
> {
>   "trailingComma": "es5",
>   "singleQuote": true
> }
> ```
> I kinda like that there's a trailing comma where we can for ES5 (Max compat) and single quotes; the default of prettier has singleQuotes turned as off and the trailing comma as none.

Updated according to given config

**Comment by didoo** [comment]
> Do you have preferences / thoughts on formatting?
No special preferences, apart from large value for wrapping lines (we all have big screens, nowadays) and trailing commas allowed.

**Comment by vvscode** [comment]
@didoo  ok. Just confirm the width ( in general I thought you have config across all the projects )

**Comment by chazzmoney** [comment]
For width, what do you guys think of 120 or 140?

**Comment by dbanksdesign** [comment]
I would say 120, any of those work but probably shouldn't go above 140

**Comment by davixyz** [comment]
120 is a good width! 80 is pretty tight honestly

**Comment by vvscode** [comment]
`printWidth` for 120 chars applied

**Comment by davixyz** [comment]
LGTM Too! 👍

**Comment by chazzmoney** [comment]
LGTM! Merging :)

**Comment by didoo** [comment]
sorry for the late reply: 120 is OK :)
